x-pack/plugins/observability_solution/apm/server/lib/connections/get_connection_stats/get_destination_map.ts

```diff
@@ -67,52 +70,78 @@ export const getDestinationMap = ({
       offset,
     });
 
+    const query: QueryDslQueryContainer = {
+      bool: {
+        filter: [
+          { exists: { field: SPAN_DESTINATION_SERVICE_RESOURCE } },
+          ...rangeQuery(startWithOffset, endWithOffset),
+          ...filter,
+          ...excludeRumExitSpansQuery(),
+        ],
+      },
+    };
+
+    const hitCountResponse = await apmEventClient.search('get_exit_span_doc_count', {
+      apm: {
+        events: [ProcessorEvent.span],
+      },
+      body: {
+        size: 0,
+        track_total_hits: true,
+        query,
+      },
+    });
+
+    const totalDocCount = hitCountResponse.hits.total.value;
+    const rawSamplingProbability = Math.min(100_000 / totalDocCount, 1);
```

Comment:
100000 is an arbitrary value. I'm not sure what would be a right balance here

Reply:
I don't know a lot on this topic but I think there is a trade-off between achieving statistical significance and maintaining manageable sample sizes for performance optimization.  The higher the number is, the higher the probability, the longer it takes?  I think the balance is going to be based on our performance expectations.

Reply:
I think statistical significance is somewhat easier than predicting performance. You can make a reasonable estimation of how many documents will be processed per second account for e.g. 3x difference both ways (faster and slower), but that will be per shard, and in bigger clusters the data might be distributed amongst shards, nodes, search threads etc. From my experience, 100k on a single shard will be fast enough, so you can probably increase that number quite a bit. For a simple terms agg you can expect about 10m docs per second, this request will be a little slower of course. I would play around a bit with the number on Serverless to see what performance you get out of it. If long tail is important, you can also do a follow-up request where you've removed the x biggest buckets, but that seems overkill.

Reply:
@neptunian @dgieselaar I played with different sampling probabilities:

Query executed:
```
GET traces-apm*,apm*/_search
{
  "track_total_hits": false,
  "size": 0,
  "query": {
    "bool": {
      "filter": [
        {
          "terms": {
            "processor.event": [
              "span"
            ]
          }
        }
      ],
      "must": [
        {
          "bool": {
            "filter": [
              {
                "exists": {
                  "field": "span.destination.service.resource"
                }
              },
              {
                "range": {
                  "@timestamp": {
                    "gte": "now-30d",
                    "lte": "now",
                  }
                }
              },
              {
                "bool": {
                  "must_not": [
                    {
                      "terms": {
                        "agent.name": [
                          "js-base",
                          "rum-js",
                          "opentelemetry/webjs"
                        ]
                      }
                    }
                  ]
                }
              }
            ]
          }
        }
      ]
    }
  },
  "aggs": {
    "sampling": {
      "random_sampler": {
        "probability": {PROBABILITY}
        "seed": 815613888
      },
      "aggs": {
        "connections": {
          "composite": {
            "size": 10000,
            "sources": [
              {
                "dependencyName": {
                  "terms": {
                    "field": "span.destination.service.resource"
                  }
                }
              },
              {
                "eventOutcome": {
                  "terms": {
                    "field": "event.outcome"
                  }
                }
              }
            ]
          },
          "aggs": {
            "sample": {
              "top_metrics": {
                "size": 1,
                "metrics": [
                  {
                    "field": "span.type"
                  },
                  {
                    "field": "span.subtype"
                  },
                  {
                    "field": "span.id"
                  }
                ],
                "sort": [
                  {
                    "@timestamp": "asc"
                  }
                ]
              }
            }
          }
        }
      }
    }
  }
}
```
Total docs: ~633.278.865:
 Scenarios: The query above, running it with `size=0` and `size=1`, to force more load on ES 

| static number |  computed probability | doc_count  |  took (` with size=1`)  |  took  |
|---------------|------------|-------------|-----------------|-------|
|  100k |  0.000157908317373 | 99.806  | 12.8s  | 3.3s |
|  500k | 0.000789541586865  |  500.187  | 12.9s  | 2.9s |
| 1M  |  0.00157908317373 |  999.821 | 12.9s  | 2.8s |
| 5M   | 0.00789541586865  |  5.004.747 |  12.8s | 3.4s |
|  10M |  0.0157908317373 | 10.009.281  |  12.4s | 3.3s |
|  25M  | 0.039477079343237  | 25.026.358  |  13.8s | 5s |
| 50M   | 0.0789541586865  |  50.141.260 |  16.4s | 7.2s |
|  100M |  0.157908317373 | 100.121.927  | 22.8s  |  10.3s |
|  150M  | 0,236862476059421  | 150.029.124   |  23.5s | 14.8s |
|  250M |   0.39477079343237 |  249.747.860 |  timeout  | 22s |


10M - good performance. It's only a small fraction of documents, but it could be enough(?)
between 25M and 50M - performance is a bit compromised but could be acceptable
Between 100M and 150M - could lead to some performance issues, but I'm not sure if it would get to cause a 502
Anything above 250M may cause timeouts